harvest/algo.py

```diff
@@ -40,6 +41,13 @@ def setup(self):
 
     def main(self):
         pass
+
+    def add_plugin(self, plugin: Plugin):
+        value = getattr(self, plugin.name, None)
+        if value is None:
+            setattr(self, plugin.name, plugin)
+        else:
+            error(f"Plugin name is already in use! {plugin.name} points to {value}.")
```

Comment:
Handles adding a plugin. All plugins must have a name attribute which will be used to access the plugin from within the algo. For example if `plugin.name = "my_plug"` then to access it you will use `self.my_plug.some_function()`